### PR TITLE
Doc: check for autogenerated ids

### PR DESCRIPTION
**Pull request by dperpeet**

### CODE REVIEW

**Comment by dperpeet** [comment]
Sample failure:
```
dist/guide/html/remotectl.8.html:69:<a name="idm139687938269984"></a><h2>OPTIONS</h2>
Unexpected generated id in the documentation
Makefile:8208: recipe for target 'dist/guide/html/index.html' failed
make[1]: *** [dist/guide/html/index.html] Error 1
make[1]: Leaving directory '/home/dev/build'
Makefile:2737: recipe for target 'all' failed
make: *** [all] Error 2
```

File: doc/guide/Makefile-guide.am [hunk 1]
```diff
@@ -98,6 +98,11 @@ dist/guide/html/index.html: $(GUIDE_DOCBOOK) $(GUIDE_INCLUDES) $(man_MANS) $(GUI
 		LANG=en_US.UTF-8 $(XMLTO) html -m $(srcdir)/doc/guide/gtk-doc.xsl -o dist/guide/html/ \
```
Comment (martinpitt) [inline doc/guide/Makefile-guide.am]: nitpick: there is no need to separate the four commands with &&, as make already runs all the commands in `set -e` style. Then the double indentation could be dropped (after the `MKDIR_P`).

File: doc/guide/Makefile-guide.am [hunk 2]
```diff
@@ -98,6 +98,11 @@ dist/guide/html/index.html: $(GUIDE_DOCBOOK) $(GUIDE_INCLUDES) $(man_MANS) $(GUI
...
 			--searchpath $(abs_builddir):$(abs_srcdir):$(abs_builddir)/doc/guide \
 			$(srcdir)/$(GUIDE_DOCBOOK)
+		if grep -n -r 'name=\"id' dist/guide/html; then \
+			echo "Unexpected generated id in the documentation"; \
```
Comment (martinpitt) [inline doc/guide/Makefile-guide.am]: I'd send the error message and grep output to stderr (`... >&2`) in case this is being run in quiet mode. We actually do this in our CI system (see my recent lintian fix), so this would swallow the error message.

File: doc/guide/Makefile-guide.am [hunk 3]
```diff
@@ -98,6 +98,11 @@ dist/guide/html/index.html: $(GUIDE_DOCBOOK) $(GUIDE_INCLUDES) $(man_MANS) $(GUI
...
+			rm dist/guide/html/index.html; \
```
Comment (martinpitt) [inline doc/guide/Makefile-guide.am]: (as discussed in meeting) Not the most elegant, but this will do and should be reasonably robust.

File: doc/man/remotectl.xml [hunk 4]
```diff
@@ -44,12 +44,12 @@
   </refsynopsisdiv>
 
 
-  <refsect1>
+  <refsect1 id="remotectl-description">
```
Comment (martinpitt) [inline doc/man/remotectl.xml]: I'm confused -- the commit message says "remove generated IDs" but this commit *adds* IDs?

File: doc/guide/Makefile-guide.am (hunk 1 above)
Comment (dperpeet) [inline doc/guide/Makefile-guide.am]: you need to refresh, I already removed that :)

File: doc/guide/Makefile-guide.am (hunk 2 above)
Comment (dperpeet) [inline doc/guide/Makefile-guide.am]: good point

File: doc/guide/Makefile-guide.am (hunk 1 above)
Comment (martinpitt) [inline doc/guide/Makefile-guide.am]: Yes, I saw -- what I meant is that while your are at touching these bits it might be nice to make a separate commit which removes the existing `&&` and the extra indentation too.

File: doc/man/remotectl.xml (hunk 4 above)
Comment (dperpeet) [inline doc/man/remotectl.xml]: Yes! We add IDs so they aren't generated during the make step. Generating IDs leads to different results, meaning the build isn't reproducible and patchsets always include these lines, even if nothing else changed.

File: doc/guide/Makefile-guide.am (hunk 1 above)
Comment (dperpeet) [inline doc/guide/Makefile-guide.am]: I added a commit to clean the entire file

**Comment by dperpeet** [comment]
Updated pr based on all previous comments, thanks @martinpitt

File: doc/guide/Makefile-guide.am [hunk 5]
```diff
@@ -98,6 +98,11 @@ dist/guide/html/index.html: $(GUIDE_DOCBOOK) $(GUIDE_INCLUDES) $(man_MANS) $(GUI
 		LANG=en_US.UTF-8 $(XMLTO) html -m $(srcdir)/doc/guide/gtk-doc.xsl -o dist/guide/html/ \
 			--searchpath $(abs_builddir):$(abs_srcdir):$(abs_builddir)/doc/guide \
 			$(srcdir)/$(GUIDE_DOCBOOK)
+		if grep -n -r 'name=\"id' dist/guide/html >&2; then \
+			echo "Unexpected generated id in the documentation"; \
```
Comment (martinpitt) [inline doc/guide/Makefile-guide.am]: `>&2` to the echo too, please?

File: doc/man/remotectl.xml (hunk 4 above)
Comment (martinpitt) [inline doc/man/remotectl.xml]: Ah, makes sense. Could you change the commit message accordingly? Like "Add fixed IDs to avoid unpredictable autogenerated ones"?

Comment (dperpeet) [inline doc/man/remotectl.xml]: I updated the commit message to make this a bit clearer.

**Comment by dperpeet** [comment]
Commit message was already changed, added the `>&2`

**Comment by martinpitt** [comment]
There are a few random test case failures here, but the main point is the build and semaphore, and that's all good.

**Comment by dperpeet** [comment]
Thanks for merging - I hope you tried out the actual guide and nothing important was removed. :)
